blocks/library/gallery/block.js

```diff
@@ -86,7 +88,7 @@ class GalleryBlock extends Component {
 		this.props.setAttributes( {
 			images: images.map( ( attributes ) => ( {
 				...attributes,
-				caption: attributes.caption ? [ attributes.caption ] : [],
+				caption: hpqParse( attributes.caption, children() ),
```

Comment:
I feel like this should be part of the attributes declaration, rather than something afterwards. Would love to have @aduth's opinion/suggestion.

Reply:
hm also how does this belong in `onSelectImages`? When is it not being parsed exactly?

Reply:
I think whatever calls `onSelectImages` on the image(s) should do the parsing?

Reply:
Hi @iseulde when we select an image from the mediaLibrary the media library pass us a string with a caption. The mediaLibrary should not handle the creation of the react children as the user of the library may, in fact, want things in string format. In order for us to correctly pass that html and use it in our captions RichText fields we need to pass it to the "react children format". That's exactly what this line is doing.

Reply:
I think that you can do it inside the `MediaUpload` component as @iseulde suggested. You would have only one place where this transformation happens and all blocks would benefit from it.

https://github.com/WordPress/gutenberg/blob/a2f41b7206a87ce56995b2a869fa2701a095866c/blocks/media-upload/index.js#L113-L117

Reply:
Yes @iseulde and @gziolo I think it may make sense to do it in MediaUpload because it is part of the blocks, I had the idea it was part of the components and so it would not be good to include things from the block parsing. I will update the code.

Reply:
Code was updated.